### DISCUSSION TITLE
Import relationship from SQL-database

### DISCUSSION THREAD

**Discussion by NogrTL** [Budibase/budibase#1827] [Ideas]
I was playing with your product. It is cool, I should say.

But creating UI from properly separated SQL tables with a lot of relationships is kinda impossible for now.
SQL-servers all have some sort of column relationships, so I guess it is possible to import it.

It would be really cool to do

**Comment by shogunpurple** [comment]
@NogrTL Thank you for the feedback.

We are actually working on this right now. The SQL table relationships will not be imported automatically, but you will be able to set up relationships between tables, and then use those relationships in your UI.

For example, if you have a `people` and `tasks` table, you could set up a  `people` to `tasks` relationship, and then render this relationship (and create records for it) in the UI.

We are hoping to have this feature shipped in the next week or two.

Thanks!

**Reply by samthompsonkennedy** [reply]
This is awesome, and I'm looking forward to it!

For the meantime, how do I setup these relationships? I'm using Postgres.

Thanks in advance!

**Reply by mike12345567** [reply]
Hi @samfroggy currently the only method to perform joins is to create a custom query and use this instead, the ad query button at the bottom of a data source allows you to do this. This allows just writing a standard SQL statement that you can perform any sort of joins within.

![image](https://user-images.githubusercontent.com/4407001/124252871-587d2e00-db1f-11eb-86c8-fa53e1901dac.png)

By next week we should have the new relationship feature fleshed out, which will allow creating relationships in the UI and querying/creating relationships without having to write any SQL.

Heres a little sneak peak of what this feature is going to look like:
![image](https://user-images.githubusercontent.com/4407001/124253430-ec4efa00-db1f-11eb-95e7-c253eb46def3.png)
